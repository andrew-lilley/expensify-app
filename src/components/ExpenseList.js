import React from "react";
import { connect } from "react-redux";
import ExpenseListItem from "./ExpenseListItem";
import selectExpenses from "../selectors/expenses";

// Export so that the component can be tested.
export const ExpenseList = (props) => (
  <div>
    <h1>Expense List</h1>
    {props.expenses.length <= 0 && <p>There are no expenses matching your search.</p>}
    {props.expenses.map((expense) => {
      return <ExpenseListItem key={expense.id} {...expense} store={props} />
    })}
  </div>
);

const mapStatetoProps = (state) => {
  return {
    expenses: selectExpenses(state.expenses, state.filters)
  };
};

// Connect component to the redux store.
export default connect(mapStatetoProps)(ExpenseList);