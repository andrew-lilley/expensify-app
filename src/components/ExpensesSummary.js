import React from "react";
import { connect } from "react-redux";
import numeral from "numeral";
import 'numeral/locales/en-gb';
numeral.locale('en-gb');
import selectExpenses from "../selectors/expenses";
import selectExpensesTotal from "../selectors/expenses-total";

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
  const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
  const formattedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00');
  return (
    <div>
      <p>Viewing {expenseCount} {expenseWord} totalling {formattedExpensesTotal}</p>
    </div>
  )
};

const mapStatetoProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);
  return {
    expenseCount: visibleExpenses.length,
    expensesTotal: selectExpensesTotal(visibleExpenses)
  };
};

// Connect component to the redux store.
export default connect(mapStatetoProps)(ExpensesSummary);