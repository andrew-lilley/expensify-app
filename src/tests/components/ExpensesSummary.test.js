import React from "react";
import { shallow } from "enzyme";
import { ExpensesSummary } from "../../components/ExpensesSummary";

test('Should correctly return count and total for a single expense', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={145} />);
  expect(wrapper).toMatchSnapshot(); 
});

test('Should correctly return count and total for multiple expenses', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={3} expensesTotal={1445} />);
  expect(wrapper).toMatchSnapshot();
});