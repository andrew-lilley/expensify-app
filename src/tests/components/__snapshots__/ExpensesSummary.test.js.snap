// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly return count and total for a single expense 1`] = `
<div>
  <p>
    Viewing 
    1
     
    expense
     totalling 
    £1.45
  </p>
</div>
`;

exports[`Should correctly return count and total for multiple expenses 1`] = `
<div>
  <p>
    Viewing 
    3
     
    expenses
     totalling 
    £14.45
  </p>
</div>
`;
